service/controller/resource/cloudconfig/desired.go

```diff
@@ -95,9 +94,33 @@ func (r *Resource) GetDesiredState(ctx context.Context, obj interface{}) (interf
 		}
 	}
 
+	organizationAzureClientCredentialsConfig, subscriptionID, _, err := r.credentialProvider.GetOrganizationAzureCredentials(ctx, credentialSecret.Namespace, credentialSecret.Name)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+	var cloudConfig *cloudconfig.CloudConfig
+	{
+		c := cloudconfig.Config{
+			Azure:                  r.azure,
+			AzureClientCredentials: organizationAzureClientCredentialsConfig,
+			CertsSearcher:          r.certsSearcher,
+			Logger:                 r.logger,
+			Ignition:               r.ignition,
+			OIDC:                   r.oidc,
+			RandomkeysSearcher:     r.randomKeysSearcher,
+			SSOPublicKey:           r.ssoPublicKey,
+			SubscriptionID:         subscriptionID,
+		}
+
+		cloudConfig, err = cloudconfig.New(c)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+	}
+
 	output := []ContainerObjectState{}
 	{
-		b, err := cc.CloudConfig.NewMasterTemplate(ctx, ignitionTemplateData, encrypter)
+		b, err := cloudConfig.NewMasterTemplate(ctx, ignitionTemplateData, encrypter)
```

Comment:
This is the relevant change. We create the cloudconfig object instead of fetching it from the `controllercontext`.